# Data Analysis on Android and iOS mobile apps

This research project is focused on all the apps that are available on Google Play and the Apple App Store. The data set contains information on a sample of at least 7200 applications which we will disect and analyse to reach the goal of this project.

We aim to provide insights on what type of apps are likely to attract more users. As the company we are working for offers its apps for free, and generates revenue through ads, it is important to understand which type of free apps are likely to generate the most traffic.

In [1]:
open('AppleStore.csv', encoding='utf8')

<_io.TextIOWrapper name='AppleStore.csv' mode='r' encoding='utf8'>

In [2]:
open('googleplaystore.csv', encoding='utf8')

<_io.TextIOWrapper name='googleplaystore.csv' mode='r' encoding='utf8'>

# Opening and Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

For the purposes of this project, we will only be exploring a sample size of the above data.

1. A data set containg data about approximately 10,000 Android apps from Google Play (the data was collected in August 2018) - [Found here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv")
2. A data set containing data about approximately 7,000 iOS apps from the App Store (the data was collected in July 2017) - [Found here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv") 


In [3]:
from csv import reader

# The Google Play data set
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

# The App Store data set
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [4]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


Exploring the android apps on the Google Play Store data set shows that we have 10,841 apps, with 13 columns of specific data about each app.

The most interesting columns for us to focus on in this research in `App`, `Category`, `Installs`, `Price` and `Genres`. 

In [5]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


 Exploring the iOS apps on the App Store data set shows that we have 7,197 apps, with 16 columns of specific data about each app.
 
 The most interesting columns for us to focus on in this research is `track_name`, `currency`, `price`, `user_rating` and `prime_genre`. More info about this data set, and column headers can be found in this data set  [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps"). 

## Data Cleaning

In this section, I will focus on cleaning the data sets that have been presented above. This is an important step in insuring analytical accuracy throughout this project.

Firstly, I will:

1. Detect inaccurate data, and correct or remove it.
2. Detect duplicate data, and remove the duplicates.

Secondly, in the research brief, our company states that they only build apps that are free to download and install. The main target audience of these applications is English-speaking. Thus, I will:

1. Remove non-English apps like 爱奇艺PPS -《欢乐颂2》电视剧热播.
2. Remove apps that aren't free.

## Removing Incorrect Data

In this [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion") dedicated to our Google Play Store data set, I have noticed an error being mentioned for row 10472. I will validate this error, and remove it if necessary.

In [6]:
print(android[10472]) #wrong data (discussion)
print('\n')
print(android_header) #header row
print('\n')
print(android[0]) #correct data

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


As shown above, there is an error in row 10472 in the data set I have used. In the column 4, indicating number of reviews, the value is `1.9`, a string which goes against the general integer format of that column in the data set.

In [7]:
del android[10472]

print(android[10472])

['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


This row has now been removed, and there is now longer an inconsistency in the format of the column position 4 in the data set.

## Removing Duplicate Data

### Part One - Identifying Duplicates

Spending more time reading over the Google Play Store data set has shown that there are duplicate entries that need to be removed.

An example of such duplicates will be shown below for the app `Instagram`:

In [8]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As shown above, there are four duplicate entries for the app `Instagram`.

I will highlight exactly how many duplicate apps appear in this data set below:

In [9]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
            unique_apps.append(name)
            
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplcate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplcate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We do not want duplicate data in our data set when we begin the analysis, because this will impact our accuracy come the end of this report.

When looking at the four entries for the app `Instagram` above, you will see that the main differentiating factor is the number of reviews, which appears in column position 4.

I will use this factor to remove the duplicates, so it is not done at random, by building a dictionary with only the app that has the most number of reviews - signifying that it was the latest entry for the app.

### Part Two - Removing Duplicates

In [10]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

Above, I showed that there are 1,181 cases of duplicate entries in this data set. So, once clean, the expected length of this data set should be the difference of the duplicate entries and the full data set of 10,841.

In [11]:
print('Expected length:', len(android) - 1181)
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


Now that all duplicates have been removed from the dictionary `reviews_max`, I will use it to build two new dictionaries:

1. `android_clean` - where all the new cleaned data set will be stored
2. `already_added` - which will just store app names

In [12]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

Just to demonstrate that the above is correct, I will show an example of the data set below, with the length to compare to the expected length above.

In [13]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


We have 9,659 rows as expected.

### Part Three - Identifying non-English Apps

When exploring the data sets for both the `android` and `ios` apps, I have noticed multiple apps that are targeted at a non-English audience. Examples below:

In [14]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


As our company only focuses on developing applications for an English audience only, we will have to remove apps from the data set that do not fit the criteria.

I will write a function that identifies chracters that are considered as non-English (as they fall outside the American Standard Code for Information Interchange guidelines of 1 - 127 being English characters)

In [15]:
def english_apps(string):
    
    for character in string:
        if ord(character) > 127:
            return False

    return True

print(english_apps('Instagram'))
print(english_apps('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_apps('Docs To Go™ Free Office Suite'))
print(english_apps('Instachat 😜'))

True
False
False
False


As seen above, the function works as expected, however it incorrectly indentifies app_names with characters like `™` and `😜` as being non-english.

Using the function as is will lead to us removing useful data from our data set.

#### Solution

The solution for this, is to assess whether there are more than 3 non-english (ASCII 0 - 127) characters in a string.

I have reformulated the function from the code above to reflect that change.

In [16]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


This works as expected, correctly labelling `Docs To Go™ Free Office Suite` and `Instachat 😜` as English apps.

In [17]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

After exploring the two data sets, and removing duplicates, incorrect data, and non-english apps, we see:

1. A total of 9,614 apps to analyse in the `android` data set
2. A total of 6,183 apps to analyse in the `ios` data set

The last step in cleaning the two data sets is to seperate the non-free apps from the free apps. 

As our company only releases free apps, and generates revenue from in-app ads, the non-free apps are irrelevant for this analysis.

In [18]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print(len(android_final))
print(len(ios_final))

8864
3222


After cleaning the data, and ensuring we are only analysing relevant data, we are left with a total of:

1. 8,864 free, english-only apps in the `android` data set
2. 3,222 free, english-only apps in the `ios` data set

This will be sufficient for our analysis.

# Data Analysis

In this section we will begin analysing the data we have cleaned previously. The goal is to identify the kids of apps that are likely to attract more users because our revenues (ad-revenue) is influenced by the number of people using our apps.

In order to minimize risks and overhead at our company when building apps, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Hence why we need to analyse the apps that would be successful on both the Google Play Store and the App Store.

## Frequency Table - Share of Apps 

Below, I will inspect the header rows of each data set to determine which columns are relevant to generate frequency tables for our analysis:

In [22]:
print(android_header)
print('\n')
print(ios_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The data I will use for the frequency tables are:

1. **Google Play Store** - the `Category` and `Genres` columns.

1. **App Store** - the `prime_genre` column.

I will build two functions to build the frequency table:

- One function to build the frequency table and show results in a percentage format.

- A second function to display the results in descending order.

In [26]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
             table[value] = 1
            
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage

    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Frequency Tables

Now that the function has been defined, I will use it to build the frequency tables for both data sets (Play Store and App Store).

Starting with the iOS apps on the App Store (free English apps only):

In [27]:
display_table(ios_final, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


The above shows that 58% of all iOS apps on the App Store are Games, the largest majority in our data set by a large margin. Followed by Entertainment with an 8% share and Photo & Video with a 5% share. Apps for specific purposes such as Education and Social Networking are both around the 3% mark. The data shows that the majority of apps on the App Store are for entertainment purposes (Games, Entertainment, Photos and Video, Sports, Music, Health & Fitness, etc). 

Next, I will apply the function to build the frequency tables for the Android apps in the Google Play Store (free English apps only):

In [30]:
display_table(android_final, 1) #category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The Google Play data set shows a slightly different picture than that of the App Store data set. Here we see 19% of all apps being under the rather generic Category `FAMILY`, with `GAME` at 9.7% and `TOOLS` at 8.4%. 

There seems to be more apps for specific purposes such as `TOOLS`, `BUSINESS` and `PRODUCTIVITY` on the Play Store than on the App Store.

However, as we can see in this image, the category `FAMILY` (which accounts for 19% of all apps on the Play Store consists mostly of games for children.
 
![alt text](https://www.lowyat.net/wp-content/uploads/2015/06/Google-Play-Store-Family-Category.jpg)

Now, for the Genres column:

In [32]:
display_table(android_final, -4) #genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

It is not abundantly clear what the difference is between the Category and Genres columns. However, the data is on a more granular level in the Genres column.

Here, we see that Tools has the most share of apps on the Play Store at 8.5%, followed by Entertainment at 6% and Education at 5.3%.

It is important to note for all the above data that a higher % of apps in a certain genre does not necessarily indicate popularity, that will be analysed in the next section. 

## Frequency Table - Most popular apps by Genre

Below, I will analyse the data that shows the popularity of apps based on interactons with the application.

- For the Android Google Play Store data set we will use the information found in the `Installs` column

- For the iOS App Store data set we will use a workaround as there is no column that clear illustrates the number of downloads. I will use the column `rating_count_tot` as a proxy for downloads (this column the number of user ratings).

## iOS App Store

Firstly, we will look into the iOS App Store data set, as I need to do the following in order to determine the number of user ratings per genre (we are using user ratings as a proxy for number of downloands):

- Isolate the apps of each genre.
- Sum up the user ratings for the apps of that genre.
- Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

In [38]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Photo & Video : 28441.54375
Music : 57326.530303030304
Health & Fitness : 23298.015384615384
Travel : 28243.8
Book : 39758.5
Utilities : 18684.456790123455
Shopping : 26919.690476190477
Medical : 612.0
Games : 22788.6696905016
Food & Drink : 33333.92307692308
Catalogs : 4004.0
Education : 7003.983050847458
Finance : 31467.944444444445
News : 21248.023255813954
Lifestyle : 16485.764705882353
Weather : 52279.892857142855
Social Networking : 71548.34905660378
Productivity : 21028.410714285714
Entertainment : 14029.830708661417
Reference : 74942.11111111111
Business : 7491.117647058823
Navigation : 86090.33333333333
Sports : 23008.898550724636


From the above data, we can see that `Navigation` apps are the most popular genre on the App Store. However, this is greatly influenced by two widely used applications, Waze and Google Maps which have nearly half a million reviews between them.

In [45]:
for app in ios_final:
    if app[-5] == "Navigation":
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


We can see the same pattern with `Social Network` apps on the iOS App Store where popular apps such as Facebook, WhatsApp Messenger, Messenger, Tumblr and Skype for iPhone have a majority of the reviews.

In [46]:
for app in ios_final:
    if app[-5] == "Social Networking":
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

As these two genres seem to be dominated by some of the major players as mentioned above, I do not suggest these two genres as focus points for developig free applications for our company.

The two genres that I have identified as being particularly interesting are the `Reference` and `Book` genres, which are not skewed by major players such as the examples above.

In [47]:
for app in ios_final:
    if app[-5] == "Reference":
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [52]:
for app in ios_final:
    if app[-5] == "Book":
        print(app[1], ':', app[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


These two genres are not saturated with too many major players, and the popularity of the genres show a possible opportunity for a new entrant.

## Google Play Store

Now we will look into the Google Play Store data set. However, before the analysis I have noticed that the format of the data is slightly different to what is needed.

In [53]:
display_table(android_final, 5) #the Installs column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


As seen above, the data consists of approximate ranges (100,000+ for example), which is okay as we do not need exact numbers to determine the most popular apps.

However, we will need to remove the `,` and `+` from the data so that it can be converted from a string to a float.

In [62]:
category_android = freq_table(android_final, 1)

for category in category_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace("+", "")
            n_installs = n_installs.replace(",", "")
            total += float(n_installs)
            len_category += 1
        
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

BOOKS_AND_REFERENCE : 8767811.894736841
FOOD_AND_DRINK : 1924897.7363636363
AUTO_AND_VEHICLES : 647317.8170731707
EVENTS : 253542.22222222222
COMMUNICATION : 38456119.167247385
WEATHER : 5074486.197183099
SPORTS : 3638640.1428571427
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
LIBRARIES_AND_DEMO : 638503.734939759
MEDICAL : 120550.61980830671
ENTERTAINMENT : 11640705.88235294
ART_AND_DESIGN : 1986335.0877192982
HOUSE_AND_HOME : 1331540.5616438356
PRODUCTIVITY : 16787331.344927534
FINANCE : 1387692.475609756
SOCIAL : 23253652.127118643
BEAUTY : 513151.88679245283
TRAVEL_AND_LOCAL : 13984077.710144928
PARENTING : 542603.6206896552
EDUCATION : 1833495.145631068
VIDEO_PLAYERS : 24727872.452830188
HEALTH_AND_FITNESS : 4188821.9853479853
FAMILY : 3695641.8198090694
BUSINESS : 1712290.1474201474
NEWS_AND_MAGAZINES : 9549178.467741935
DATING : 854028.8303030303
GAME : 15588015.603248259
COMICS : 817657.2727272727
LIFESTYLE : 1437816.2687861272
PHOTOGRAPHY : 17840110.402298

Looking at the data above, there are a few genres that stand out in terms of downloads. However, the major stand out number is that of `COMMUNICATION` with 38,456,119 downloads. Diving into this genre shows a dominance of a few major players, which significantly skews the cohort.

In [63]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

With the likes of WhatsApp Messenger, Messenger, Skype, Google Chrome, Gmail and Hangouts all having over 1,000,000 installs, it inflates the average install per application by a significant margin.

These are massive players in their respective markets that will be difficult to compete against.

Further investigating the most popular categories above, I have identified the category `BOOKS_AND_REFERENCE` with an average of 8,767,811 installs per application. The distribution of these installs is as follows:

In [66]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

This genre has a clearer distribution of installs across the data, with there being only handful of applications with over 1,000,000 installs:

In [67]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


This genre presents an opportunity to the company as there are fewer large players preventing new entrants into the market, and the average number of installs per application is high.

Further, it seems that this niche market is dominated by e-Readers and digital versions of dictionaries. So I advised against creating either of those two types of apps as there is considerable competition already.

However, digital versions of popular books such as the Bible, the Quran and other popular books have an even ditribution of downloads across the niche. I would advise creating an app that is a digital version of a newer popular book, but with additional features that sets it apart from the rest of the competition.

# Conclusion

In conclusion, this project's focus was to analyse apps across the iOS App Store and the Google Play Store with the goal of identifying the most profitable genre for free English-targeted applications.

I have advised on creating an app that is a digital version of a popular book, but with additional features that sets it apart from the rest of the competition. This app would prove popular in both app stores based on the data set we have analysed, with the Book and Reference genre being popular in both stores.